### DISCUSSION TITLE
Solution for Github.com’s use of the async clipboard API on Web

### DISCUSSION THREAD

**Discussion by stan-janssen** [elementary/triage#201] [Q&A] [answered]
So Github.com is broken on GNOME Web for the past weeks, as they are using the async clipboard API and Web does not play nice. There is a workaround that involves placing the following lines of code in the user-javascript.js file:

```javascript
navigator.clipboard = {};
```

[According to](https://gitlab.gnome.org/GNOME/epiphany/-/issues/2003#note_1702940) the GNOME Web developers, this problem is not solved by updating GNOME Web, but rather by updating the system-level library `WebKitGTK` to a version that is built with `webkitgtk-2.40`.

As Ubuntu does not provide that version in its Jammy repositories, is there anything else we can do in the meantime? Also: is it true that the flatpak version of GNOME Web does not include its own copy of libwebkit2gtk?

**Comment by danirabbit** [comment, marked as answer]
In my testing it did seem to be fixed by building Web against the newer runtime. I have a branch for Web 44 and a new runtime 7.2 is ready to be released. We're fixing some storage issues on our Flatpak server and then we should be able to release that within the next couple days I think!

In the meantime if you wanted you could manually build Web for your personal use by modifying this branch: https://github.com/elementary/browser/pull/83

**Reply by danirabbit** [reply]
Branch is building again and it looks like WebKitGtk 2.40 is included! Posted this comment from Web 44 :)

**Reply by stan-janssen** [reply]
Just built and ran the new version. Can confirm it works and the specific problem in Github.com is resolved. 

Is it just me though or does the new version feel a bit "off" somehow? The inertial scrolling on the trackpad of my Starbook is different; the page slows done quicker after I stop scrolling. Also: videos in Github don't seem to play anymore (like these: https://github.com/elementary/wingpanel/pull/481).

Is this something you noticed as well?

**Comment by danirabbit** [comment]
Closing as outdated since this was an issue in an older version of WebKitGTK
